main.go

```diff
@@ -188,6 +190,31 @@ func updateWatchdogTimer(a *attiny) {
 	}
 }
 
+func updateWifiState(a *attiny) {
+	connectedState := false
+	for {
+		if connectedState != wifiConnected() {
+			if err := a.SetWifiState(!connectedState); err != nil {
```

Comment:
There's a risk of concurrent writes to the attiny if there's multiple goroutines which do this. You might need to use locks or a channel to serialise access to the hardware.

Reply:
The attiny uses Mutex for stopping concurrent writes https://github.com/TheCacophonyProject/attiny-controller/blob/master/attiny.go#L154

Reply:
Ok, that's fine then as long as each logical interaction with the attiny is one read/write (which I think it is at the moment?).

Reply:
Yep. Some things like reading the battery voltage takes 2 interactions but it doesn't matter if somethings happens between them.